include/mqt-core/dd/Operations.hpp

```diff
@@ -205,54 +160,53 @@ qc::MatrixDD getDD(const qc::Operation* op, Package<Config>& dd,
     return id;
   }
 
-  if (op->isStandardOperation()) {
-    const auto* standardOp = dynamic_cast<const qc::StandardOperation*>(op);
-    const auto& targets = permutation.apply(op->getTargets());
-    const auto& controls = permutation.apply(op->getControls());
+  if (op.isStandardOperation()) {
+    const auto* standardOp = dynamic_cast<const qc::StandardOperation*>(&op);
+    const auto& targets = permutation.apply(op.getTargets());
+    const auto& controls = permutation.apply(op.getControls());
 
     if (qc::isTwoQubitGate(type)) {
       assert(targets.size() == 2);
```

Comment:
I think this distinction on whether it is a one or two target gate is superfluous based on your changes in previous commits.